js/lex.go

```diff
@@ -33,6 +33,30 @@ const (
 	TemplateToken
 )
 
+// TokenState determines a state in which next token should be read
+type TokenState uint32
```

Comment:
Not when you run a 32-bit machine, but `uint` takes the regular int size AFAIK. So on a 64-bit machine it is a `uint64`.